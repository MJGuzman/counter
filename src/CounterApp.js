import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const Increase = (e) => setCounter(counter + 1);

  const Decrease = () => {
    if (counter !== 0) {
      setCounter(counter - 1);
    }
  };

  const Reset = () => setCounter(0);

  return (
    <div>
      <h1>Contador</h1>
      <h2>{counter}</h2>
      <div>
        <button type="button" onClick={Increase}>
          Incrementar
        </button>
        <button type="button" onClick={Reset}>
          Reset
        </button>
        <button type="button" onClick={Decrease}>
          Decrease
        </button>
      </div>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
