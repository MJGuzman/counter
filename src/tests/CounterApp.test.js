import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Prueba en el componente del contador", () => {
  test("Debe de mostrar el componente correctamente", () => {
    const wrapper = shallow(<CounterApp value={10} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el valor por defecto 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    wrapper.find("h2").text();

    expect(wrapper.find("h2").text().trim()).toBe("100");
  });

  test("debe de incrementar el contador", () => {
    const wrapper = shallow(<CounterApp value={10} />);

    wrapper.find("button").at(0).simulate("click");
    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("11");
  });

  test("debe de decrementar el contador", () => {
    const wrapper = shallow(<CounterApp value={10} />);

    wrapper.find("button").at(2).simulate("click");
    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("9");
  });
});
