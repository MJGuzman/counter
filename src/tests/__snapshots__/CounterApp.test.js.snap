// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente del contador Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    Contador
  </h1>
  <h2>
    10
  </h2>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Incrementar
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Decrease
    </button>
  </div>
</div>
`;

exports[`Prueba en el componente del contador Debe de mostrar el valor por defecto 100 1`] = `
<div>
  <h1>
    Contador
  </h1>
  <h2>
    10
  </h2>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Incrementar
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Decrease
    </button>
  </div>
</div>
`;
